### ISSUE TITLE
segment_from_mask with logits does not work for non-square shapes

### ISSUE THREAD

**Issue by constantinpape** [computational-cell-analytics/micro-sam#5]
See https://github.com/computational-cell-analytics/micro-sam/blob/master/micro_sam/segment_from_prompts.py#L52.

**Comment by constantinpape** [comment]
Need to reverse engineer the padding here: https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/modeling/sam.py#L151-L162
